### PR TITLE
Update plugin parent POM and BOM

### PR DESCRIPTION
**Pull request by basil**

### CODE REVIEW

**Comment by basil** [comment]
@jglick A release of this would facilitate Java 17 PCT testing.

File: pom.xml [hunk 1]
```diff
@@ -14,28 +14,32 @@
...
-        <artifactId>bom-2.249.x</artifactId>
-        <version>961.vf0c9f6f59827</version>
+        <artifactId>bom-2.289.x</artifactId>
+        <version>1289.v5c4b_1c43511b_</version>
         <scope>import</scope>
         <type>pom</type>
       </dependency>
+      <dependency>
+        <groupId>org.apache.commons</groupId>
+        <artifactId>commons-lang3</artifactId>
```
Comment (jglick) [inline pom.xml:40]: Mm, reminds me that this really needs to be made into a library wrapper plugin.

**Review by jglick — APPROVED** [review]
Thanks!

File: pom.xml (hunk 1 above)
Comment (timja) [reply pom.xml:40]: https://github.com/jenkinsci/commons-lang3-api-plugin ?

**Comment by jglick** [comment]
https://github.com/jenkinsci/ant-plugin/releases/tag/473.vcfd6b_16ca_581

**Comment by basil** [comment]
Thank you!

**Pull request merged**